pkg/kubernetes/namespace.go

```diff
@@ -2,39 +2,71 @@ package kubernetes
 
 import (
 	"fmt"
+	"strings"
 
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclient "k8s.io/client-go/kubernetes"
 )
 
-// Namespace creates a corev1.Namespace object with the standard labels set
-// using the provided name.
-func Namespace(name string) *corev1.Namespace {
+// NewNamespace creates a corev1.Namespace object using the provided name.
+func NewNamespace(name string) *corev1.Namespace {
 	return &corev1.Namespace{
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "Namespace",
 			APIVersion: corev1.SchemeGroupVersion.String(),
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:   name,
-			Labels: map[string]string{"name": name},
+			Name: name,
 		},
 	}
 }
 
-const namespaceTemplate = `---
-apiVersion: v1
-kind: Namespace
-metadata:
-  labels:
-    name: %s
-  name: %s
-`
-
-// NamespaceYAML returns a YAML string for a Kubernetes Namespace object with
-// the standard labels set using the provided name.
+// NewNamespaceYAML returns a YAML string for a Kubernetes Namespace object.
 // N.B.: Kubernetes' serializers are not used as unnecessary fields are being
 // generated, e.g.: spec, status, creatimeTimestamp.
-func NamespaceYAML(name string) []byte {
-	return []byte(fmt.Sprintf(namespaceTemplate, name, name))
+func NewNamespaceYAML(name string) []byte {
+	nsFmt := strings.Join(
+		[]string{
+			"---",
+			"apiVersion: v1",
+			"kind: Namespace",
+			"metadata: {name: %s}",
+		},
+		"\n")
+
+	return []byte(fmt.Sprintf(nsFmt, name))
```

Comment:
Any reason for not using `yaml.Marshal` from `sigs.k8s.io/yaml` here? Then we could do:

```golang
func NewNamespaceYAML(name string) ([]byte, error) {
    return yaml.Marshal(NewNamespace(name))
}
```

This would be more type-safe.

Reply:
It's in the comment - we want to output a clean object without `creationTimestap: nil`, as it will be written out to user's git repo and they might end up editing it or at least looking at it.

Reply:
Ah, I missed that comment since it wasn't introduced in this PR. Thanks for clarifying.